src/main/java/use_cases/add_edit_category_use_case/CategoryUCI.java

```diff
@@ -0,0 +1,153 @@
+package use_cases.add_edit_category_use_case;
+
+import entities.Category;
+import entities.EntityException;
+import entities.MonthlyStorage;
+
+import java.util.ArrayList;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+/**
+ * The CategoryUCI adds/creates new category, edits an existing category, and updates MonthlyStorage.categoryData.
+ * Implements CategoryIB.
+ * Contains a helper method isValidDouble() and findCategory().
+ * Creates CategoryOD objects related to specific use case fail/success conditions.
+ */
+
+public class CategoryUCI implements CategoryIB {
+    final CategoryOB categoryOB;
+
+    /**
+     * Constructs CategoryUCI.
+     * @param categoryP presenter that is related to the use case.
+     */
+    public CategoryUCI(CategoryOB categoryP) {
+        this.categoryOB = categoryP;
+    }
+
+    /**
+     * Helper method returns a double to check if the input Object is a valid double.
+     * @param value a user input
+     * @return double converted from value
+     */
+    public static double toDouble(Object value) throws NumberFormatException, NullPointerException{
+            return Double.parseDouble(String.valueOf(value));
+    }
+    /**
+     * Overrides method in CategoryIB.
+     * Attempts to add a category with information from CategoryID and returns a CategoryOD indicating whether fail/success after execution.
+     * Provides detailed fail messages according to each condition listed below:
+     *  1. NumberFormatException: User tries to add a new budget value that can not be converted to a double.
+     *  2. Category budget less than 0: User tries to add a new budget value that is a negative number.
+     *  3. EntityException: User tries to add an invalid Category name but failed. (See entities/EntityException.java)
```

Comment:
The Javadoc is a bit long and hard to read. Maybe move these bullet points to where you are wrote the code.